src/wallet/walletdb.cpp

```diff
@@ -832,6 +839,9 @@ DBErrors WalletBatch::LoadWallet(CWallet* pwallet)
                     // Set tx_corrupt back to false so that the error is only printed once (per corrupt tx)
                     wss.tx_corrupt = false;
                     result = DBErrors::CORRUPT;
+                } else if (wss.descriptor_corrupt) {
+                    pwallet->WalletLogPrintf("Error: Corrupt wallet descriptor found in wallet %s. This can be fixed by removing wallet from the settings so it's not loaded at startup.\n", pwallet->GetName());
```

Comment:
Probably should export message via `strErr`

Reply:
yeah, briefly thought about it as well and.. by doing it, remove code duplication that comes with it.  But.. as I have tendencies of wanting to change too much in single PRs was balancing on whether that should be here or not.
Let me see how it will looks like.

Reply:
`strErr` is already used here, it doesn't need `util::Result`

Reply:
Yes, this should just use `strErr`. Any error put there should already make it out to the user on the RPC or trigger an error dialog in the GUI.

Reply:
`strErr` isn't propagated to the GUI nor RPC. Is internal to `WalletBatch::LoadWallet`, declared prior to call `ReadKeyValue` and only used to log the error at the end of every record read while we walk through all the db records.

To propagate the error to the upper layers need to add a ref string into the function signature (or use the result class and kill two birds with the same stone, cleaning more code in the way).

If you check the line below this logging, I’m returning right away, not letting the process continue reading all the db records. The reason is that you can have a corrupted descriptor that has an “active descriptor“ entry, which will crash the wallet once the process calls to `LoadActiveScriptPubKeyMan` (on the [map.at](http://map.at/)(id) line) as the descriptor which the active entry points was never added to the wallet.

This is something that was going to push later but essentially you can crash the wallet at startup by adding an `ACTIVEEXTERNALSPK` or `ACTIVEINTERNALSPK` record to the db that points to a random spkm id.

Reply:
going to prepare the changes and test cases so limitations of the current sources and the diff benefits are visible. The end result will be similar to what was done in #25308.

Reply:
Extra note: checked how the util::result connection would look like and changes aren't straightforward to do. It needs further code reorganizations as `DBErrors` enum mixes errors, warnings and the good result (I think that the best would be to decouple this enum into two enums errors and warnings or.. even get rid of the enum completely like it was done in #25308. but.. will leave that for another PR).

So, options:

1) If move forward by adding the ref string arg here. The same error string will be propagated to the upper layers, removing one line of code duplication but having to add the std::string ref arg in all the `WalletBatch::LoadWallet` calls (which we do have a good number of them).

2) Leave it as is now, which essentially means to act the same as with all the corrupted wallet errors: having the string duplicated inside `WalletBatch::LoadWallet` and its caller `CWallet::Create` (`wallet.cpp`). And implement the proper changes all at once in a follow-up PR.

Personally, I'm inclined to go with option (2) as it's inlined with what we currently have and doesn't introduce changes that will most likely be removed later.
Thoughts?

Reply:
sounds good, let's continue with option 2 then.